src/librekey/key_store_g10.c

```diff
@@ -441,6 +475,308 @@ parse_seckey(pgp_keydata_key_t *keydata, s_exp_t *s_exp, pgp_pubkey_alg_t alg)
     return true;
 }
 
+static pgp_seckey_t *
+g10_decrypt_seckey(const pgp_key_t *key, FILE *passfp)
+{
+    uint8_t        derived_key[PGP_MAX_KEY_SIZE];
+    char *         passphrase;
+    char           pass[MAX_PASSPHRASE_LENGTH];
+    unsigned       keysize;
+    uint8_t *      decrypted;
+    pgp_seckey_t * seckey;
+    s_exp_t        s_exp = {};
+    size_t         output_written = 0;
+    size_t         input_consumed = 0;
+    botan_cipher_t decrypt;
+
+    if (key->key.seckey.encrypted_len <= 0) {
+        fprintf(stderr, "Hasn't got encrypted data!\n");
+        return NULL;
+    }
+
+    if (pgp_getpassphrase(passfp, pass, sizeof(pass)) == 0) {
+        pass[0] = '\0';
+    }
```

Comment:
@flowher to be honest: this function [never return false](https://github.com/riboseinc/rnp/blob/bb010474d9b4a710af58cfef775b5147be9b49a6/src/lib/reader.c#L145) and it is a copy-and-paste from [here](https://github.com/riboseinc/rnp/blob/7d06d4569e458b49c7080a1ad9959dcf2c6f8f87/src/lib/pgp-key.c#L320)